io_u.c

```diff
@@ -661,9 +729,9 @@ int io_u_quiesce(struct thread_data *td)
 
 static enum fio_ddir rate_ddir(struct thread_data *td, enum fio_ddir ddir)
 {
-	enum fio_ddir odir = ddir ^ 1;
-	long usec;
-	uint64_t now;
+	enum fio_ddir min_odir;
+	long usec, now, min_time;
```

Comment:
Experience says keeping now in "only" a long will lead to unsigned integer overflow on some platforms: https://github.com/axboe/fio/commit/90eff1c9f01d2f8b4ff8bc75a2bf120a1789b37c .